core/trino-main/src/main/java/io/trino/operator/scalar/timestamp/SequenceIntervalDayToSecond.java

```diff
@@ -52,8 +51,7 @@ public static Block sequence(
 
         checkValidStep(start, stop, step);
 
-        int length = toIntExact((stop - start) / step + 1L);
-        checkMaxEntry(length);
+        int length = checkMaxEntry((stop - start) / step + 1L);
```

Comment:
It seems that you identified a couple of other overflow candidates. Would you mind filing an issue about potential overflows in `SequenceIntervalDayToSecond` and `SequenceIntervalYearToMonth`?

Reply:
Sure, filed https://github.com/trinodb/trino/issues/16894
Should I link the issue as well, or remove the change in this pr and separate it.